### PR TITLE
Adding tool csvstoskyui

### PR DESCRIPTION
**Pull request by eranra**
Tool to convert CSV files to Json file to be consumed by Skydive-UI
-==--=-=-=

Main usage is to create mocks and show static typologies using Skydive-UI

### CODE REVIEW

File: tools/csvstoskyui/csvstoskyui.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import csv
+import json
+import re
+import argparse
+
+"""
+The content of topologyrules.conf file includes a python dictionary with the following keys: 
+
+filename    - The name of the file to read the csv information from
```
Comment (hunchback) [inline tools/csvstoskyui/csvstoskyui.py]: s/from example/for example/

File: tools/csvstoskyui/csvstoskyui.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+              example: "netbox/netbox_sites.csv"
+show        - Boolean that enables or disable usage of this csv source
```
Comment (hunchback) [inline tools/csvstoskyui/csvstoskyui.py]: s/disable/disables/

**Comment by hunchback** [comment]
you are still using `config.js` while new code base using `Config.ts`

File: tools/csvstoskyui/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# skynet
+
+## Tool to visualize topology from CSVs Using Modern Skydive UI
+- - - -
+
+![](https://github.com/skydive-project/skydive-ui/blob/master/tools/csvstoskyui/images/readme_screenshot.jpg?raw=true)
```
Comment (hunchback) [inline tools/csvstoskyui/README.md]: Better if you display all edges created

You are missing Layers (all within 'Not Classified')

Comment (hunchback) [reply tools/csvstoskyui/README.md]: I would also like a screenshot of the data center example present

File: tools/csvstoskyui/csvstoskyui.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                for filterkey, filtervalue in netboxtopologyfiles[key]["valuesfilter"].iteritems():
+                    try:
+                        if not re.search(filtervalue, raw[filterkey]):
+                            filter_value = True
+                    except TypeError:
+                        filter_value = True
+                if filter_value:
+                    continue
+            if "valuesmappingcopy" in netboxtopologyfiles[key]:
+                for mapkey, mapvalue in netboxtopologyfiles[key]["valuesmappingcopy"].iteritems():
```
Comment (hunchback) [inline tools/csvstoskyui/csvstoskyui.py]: you should check (or assert) that its' a map

File: tools/csvstoskyui/csvstoskyui.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+                            match = True
+                            for idx, fromvalue in enumerate(fromvalues):
+                                tovalue = tovalues[idx]
+                                if tovalue not in dst["Metadata"] or fromvalue not in src["Metadata"] or dst["Metadata"][
+                                    tovalue] != \
+                                        src["Metadata"][fromvalue]:
+                                    match = False
+                            if match:
+                                jsondump = {"Metadata": {"RelationType": relationType},
+                                            "ID": (str("Edge-" + dst["ID"] + src["ID"] + fromtype + totype)).encode("hex"),
```
Comment (hunchback) [inline tools/csvstoskyui/csvstoskyui.py]: dst["ID"] + src["ID"] is already unique (no need to add types)

Comment (hunchback) [reply tools/csvstoskyui/csvstoskyui.py]: also not need to encode as hex

File: tools/csvstoskyui/data/example_datacenter/config.js [hunk 6]
```diff
@@ -0,0 +1,407 @@
+var config = {
```
Comment (hunchback) [inline tools/csvstoskyui/data/example_datacenter/config.js]: Import default config and then override *only* the values you have changed

File: tools/csvstoskyui/csvstoskyui.py (hunk 1 above)
Comment (eranra) [reply tools/csvstoskyui/csvstoskyui.py]: Not following you... text is saying "read from" so I do not think it sound be change to "read for". Please explain more.

File: tools/csvstoskyui/csvstoskyui.py (hunk 2 above)
Comment (eranra) [reply tools/csvstoskyui/csvstoskyui.py]: fixed

File: tools/csvstoskyui/README.md (hunk 3 above)
Comment (eranra) [reply tools/csvstoskyui/README.md]: Look on the new version with scale ... (new PR with fixes)

File: tools/csvstoskyui/csvstoskyui.py (hunk 4 above)
Comment (eranra) [reply tools/csvstoskyui/csvstoskyui.py]: This is correct to many fields and sub-fields ... need a 'project' to check the scheme ... something like https://pypi.org/project/jsonschema/ maybe I will do that sometime in the future ... but I do not have the time at this point to do that. agree with the comment.

File: tools/csvstoskyui/README.md (hunk 3 above)
Comment (eranra) [reply tools/csvstoskyui/README.md]: fixed (moved to scale view)

File: tools/csvstoskyui/csvstoskyui.py (hunk 5 above)
Comment (eranra) [reply tools/csvstoskyui/csvstoskyui.py]: Thanks, fixed

File: tools/csvstoskyui/data/example_datacenter/config.js (hunk 6 above)
Comment (eranra) [reply tools/csvstoskyui/data/example_datacenter/config.js]: Possible enhancement ... not for this cycle (maybe after we move to .ts file)

**Pull request merged**